### PR TITLE
fix(types): preserve input stream type

### PR DESCRIPTION
**Pull request by ofrobots**
Fix type definition to communicate that the stream type doesn't change.

### CODE REVIEW

**Review by stephenplusplus — APPROVED** [review]

File: index.d.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import { Stream } from "stream";
-declare function StreamEvents(stream: Stream): Stream;
+declare function StreamEvents<StreamType>(stream: StreamType): StreamType;
```
Comment (JustinBeckwith) [inline index.d.ts]: Would be nice to do `(stream: StreamType extends Stream)` to show that `StreamType` must be a `Stream`.

**Review by JustinBeckwith — APPROVED** [review]

File: index.d.ts (hunk 1 above)
Comment (ofrobots) [reply index.d.ts]: Good suggestion 👍. Done.

**Comment by ofrobots** [comment]
@stephenplusplus would you mind merging and releasing this when you get a chance? thxlotsbye.

**Comment by ofrobots** [comment]
Gentle ping 🛎

**Comment by stephenplusplus** [comment]
Done as 1.0.5. Sorry for the lag!

**Pull request merged**